commerce/domain/orchestrators/classes/CartCalculateSampleUnitTest.cls

```diff
@@ -109,12 +115,30 @@ global class CartCalculateSampleUnitTest {
         assertUnexpectedCalculations(cart, new List<String>{CART_REPRICED, INVENTORY_CHECKED, SHIPPING_RECALCULATED, POST_SHIPPING_COMPLETED, TAXES_RECALCULATED});
     }
 
+    @IsTest
+    public static void shouldRunAllCalculatorsWhenBuyerAddsCouponAtCheckout() {
```

Comment:
we don't run pricing when coupon is added/removed. Should this method be renamed?

Reply:
we do when buyerActions.isCheckoutStarted() though which is what isCouponAppliedInCheckout checks for!

Reply:
Pricing is run for start checkout, but not for when coupon is applied.

Reply:
ah right but for these unit tests, we can't manually set cart status to CHECKOUT, so as a workaround we're checking for buyerActions.isCheckoutStarted() as well to determine if CouponAppliedInCheckout.  (And pricing is run when checkout is started).

Maybe this is too hack-y though if isCheckoutStarted and isCouponChanged can't be true at the same time. In that case we might need to figure out how to mock cart status in another way.

Reply:
Since this is going public, let's look at other ways to keep it close to the real production scenarios (one buyer action at a time)

Reply:
Agreed. Let's try creating a Cart with the status set to `CHECKOUT` @smahbubani99 . If we see this works, let's update this code in the `Core` repo as well just to keep things in sync.

Reply:
Is this still the right name for this method `shouldRunAllCalculatorsWhenBuyerAddsCouponAtCheckout ` ? It shouldn't be running pricing calculator.